contracts/Provider/BasicProvider.sol

```diff
@@ -3,8 +3,9 @@ pragma solidity ^0.8.0;
 
 import "./ProviderModifiers.sol";
 import "../ProviderInterface/IProvider.sol";
+import "../ProviderInterface/IProviderSingleIdRegistrar.sol";
 
-abstract contract BasicProvider is IProvider, ProviderModifiers {
+abstract contract BasicProvider is IProvider, IProviderSingleIdRegistrar, ProviderModifiers {
```

Comment:
Why this change?

Reply:
`BasicProvider` implements the `registerPool` function. `IProviderSingleIdRegistrar` declares `registerPool`